src/Interpreters/Cache/FileSegment.cpp

```diff
@@ -339,30 +339,31 @@ void FileSegment::write(const char * from, size_t size, size_t offset)
     if (!size)
         throw Exception(ErrorCodes::LOGICAL_ERROR, "Writing zero size is not allowed");
 
-    const auto file_segment_path = getPath();
-
     {
         auto lock = lockFileSegment();
-
         assertIsDownloaderUnlocked("write", lock);
         assertNotDetachedUnlocked(lock);
+    }
+
+    const auto file_segment_path = getPath();
 
+    {
```

Comment:
why this block (l. 351-379) in indented btw?

Reply:
For readability (in my opinion). This block contains only safety & correctness checks and it seemed to look good to somehow visualize it separately.. if it makes sense..